services:
  bp-fe-zakaznik:
    image: localhost:5000/bp-fe-zakaznik:latest
    build:
      context: .
      dockerfile: frontend/zakaznik/Dockerfile
    ports:
      - "8080:80"
  bp-fe-admin:
    image: localhost:5000/bp-fe-admin:latest
    build:
      context: .
      dockerfile: frontend/admin/Dockerfile
    ports:
      - "8888:80"
    environment:
      - NODE_ENV=development
      - VITE_API_URL=http://server.bpapp.local
  bp-database:
    image: localhost:5000/bp-database:latest
    build:
      context: .
      dockerfile: database/Dockerfile
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=bp
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
  bp-server:
    image: localhost:5000/bp-server:latest
    build:
      context: .
      dockerfile: server/Dockerfile
    ports:
      - "3000:3000"
    environment:
      - NODE_ENV=development
      - DATABASE_URL=postgresql://postgres:postgres@bp-database-service:5432/bp
    depends_on:
      - bp-database

volumes:
  postgres_data: 
